### PR TITLE
MVP guidelines for CodeEditor

### PR DESCRIPTION
**Pull request by kpoulincoveo**
New guidelines
Once this version is approved and proof-read, I will based the JSON editor guidelines on it.

### Proposed Changes

New guidelines

### Potential Breaking Changes

None

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

**Review by jlevesquecoveo — APPROVED** [review]

File: packages/website/docs/CodeEditor.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+Help text should explain how or where this information can be used or to provide external references about the syntax to use. 
+Help text should be **short, preferably on one line**.
+
+Placeholder provides a **temporary example** of the syntax when the input is empty. Only use a placeholder if a short example can prevent the user from having to refer to the external resources to get started.
+
+## Feedback and Validation
+
+Code editor supports syntax formatting and basic validation of their associated language.
+
+Most of the error validation on edited code only occurs once the user saves, using the proper error messaging method is important to help users troubleshoot their own errors.
```
[the remark this answers is not in the document]
Comment (kpoulincoveo) [reply packages/website/docs/CodeEditor.md:26]: We intend to have a guideline on that soon. I'm talking about the type of component (in Toast, errors message below a input and info box) and the way to write messages (I think you have similar guidelines in your copywriting guidelines)

File: packages/website/docs/CodeEditor.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+Consider providing links to troubleshooting documentation in the external references in error messages when appropriate.
+
+If possible, consider adding a way for the user to test his code before moving on through his journey. For instance, adding a button to test a script before saving it is a good practice.
+
+## Related Components
+
+If your use case doesn't match the guidelines above, consider using one of the following components instead:
+
+-   [Code editor](#/form/JSONEditor) if the user needs to input code using JSON syntax.
```
[the remark this answers is not in the document]
Comment (kpoulincoveo) [reply packages/website/docs/CodeEditor.md:35]: We do have one for JSON. I made a mistake 🤦

File: packages/website/docs/CodeEditor.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jlevesquecoveo) [reply packages/website/docs/CodeEditor.md:26]:
```suggestion
Most of the code validation takes place once changes are saved. Writing proper error messages is important to help users troubleshoot their own errors.
```

[the remark this answers is not in the document]
Comment (jlevesquecoveo) [reply packages/website/docs/CodeEditor.md:26]: @kpoulincoveo I'm done!

**Review by jlevesquecoveo — APPROVED** [review]

**Review by aiheon — APPROVED** [review]

**Pull request merged**